homework/shared_ptr/weak_ptr.hpp

```diff
@@ -37,8 +38,21 @@ void weak_ptr<T>::checkAndDeletePointers() {
     if (!counter_->getWeakRefs() && !counter_->getRefs()) {
             delete counter_;
         }
+    // 
 }
 
+template <typename T>
+void weak_ptr<T>::checkWeakAndDeletePointers() {
+    if (counter_) {
+        counter_->decreaseWeakRefs();
+        if (counter_->getRefs() == 0 && counter_->getWeakRefs() == 0) {
+            delete counter_;
+        }
+    }
+}
```

Comment:
as well as this. It's basically duplicating `checkAndDeletePointers` and I want counter decrease to be explicitly called